### DISCUSSION TITLE
What happens if I use grad_accumulate and auto_scale_lr simultaneously?

### DISCUSSION THREAD

**Discussion by MGAMZ** [open-mmlab/mmengine#1095] [Q&A]
There are two features provided by mmengine, namely auto_scale_lr and grad_acculumate.

As designed, the actual learning rate should be determined according to batch_size of dataset and base_batch_size in auto_scale_lr.

But when I use mmengine cfg param accumulative_counts, I observe no influence on actual learning_rate.

So does this mean auto_scale_lr only considers the batch_size of input dataset regardless of a potential grad_accumulative?

I believe this may induce some potential mis-set on learning_rate and more confuse.
